docs/001_develop/03_client-capabilities/023_custom-components/03_state-management

```diff
@@ -0,0 +1,342 @@
+`;
+
+@customElement({
+  name: 'my-button',
+  template,
+  styles,
+})
+
+export class MyButton extends GenesisElement {
+    @attr label = 'Click Me';
+    # Here we are binding the text inside the button to the `label` attribute.
+});
+```
+
+#### Usage in HTML
+
+```shell
+<my-button label="Cancel"></my-button>
+```
+
+- Here, the label attribute binds directly to the button's text via `${x => x.label}`.
+- If label is set to "Cancel" in HTML `(<my-button label="Cancel">)`, the button will display "Cancel".
+- When the label value is changed dynamically, the button text will automatically update.-
+
+### Interpolated Attribute Values
+
+In cases where an attribute value is partially static and partially dynamic, we can combine static and dynamic parts using interpolated expressions. This approach is useful for creating customized button labels or CSS classes.
+
+Suppose we want to add a counter to the button label that shows the number of times it has been clicked. We could use an expression to combine the static text ("Clicked: "), and a dynamic value, such as the clickCount property:
+
+```shell
+import { attr, customElement, GenesisElement, html } from '@genesislcap/web-core';
+
+const template = html<MyButton>`
+    <button>${x => `${x.label} (Clicked: ${x.clickCount} times)`}</button>
+`;
+
+@customElement({
+  name: 'my-button',
+  template,
+  styles,
+})
+
+export class MyButton extends FASTElement {
+    @attr label = 'Click Me'; // Bind to 'label' attribute
+    clickCount = 0;
+
+    handleClick() {
+        this.clickCount++;
+    }
+}
+```
+
+### Boolean Attribute Binding
+
+Boolean attributes are special because their presence/absence determines their value, rather than their string value. In addition to the `@attr` option, an attribute can also take a configuration with the `mode` option.
+
+If the attribute is a boolean and the mode is set to "boolean" this allows GenesisElement to add/remove the attribute from the element
+
+If a boolean attribute is present in the HTML markup (e.g., `<button disabled></button>`), it is interpreted as true.
+If the attribute is absent (e.g., `<my-button></my-button>`), it is interpreted as false.
+
+We can use a ? prefix to handle boolean attributes so they’re added or removed from the DOM based on a condition.
+
+> The disabled attribute controls whether the button is clickable. Using `?disabled=${x => x.isDisabled}`, we can bind it to the component’s `isDisabled` property and dynamically control whether the button is enabled or disabled.
+
+```shell
+import { attr, customElement, GenesisElement, html } from '@genesislcap/web-core';
+
+const template = html<MyButton>`
+    <button ?disabled="${x => x.isDisabled}" @click="${x => x.handleClick}">
+        ${x => x.label}
+    </button>
+`;
+
+@customElement({
+  name: 'my-button',
+  template,
+  styles,
+})
+
+export class MyButton extends FASTElement {
+    @attr label = 'Click Me';
+    @attr({ mode: 'boolean' }) isDisabled = false;
```

Comment:
There is a slight issue in this example where you're referring to the attribute in the "Usage in HTML" section as `disabled`, but it would be `isdisabled` without the suggested change. 

Perhaps a good time to explain about the `attribute: ` key on the decorator? 😄 
```suggestion
    @attr({ mode: 'boolean', attribute: 'disabled' }) isDisabled = false;
```